frontend/components/MediaEdit.vue

```diff
@@ -0,0 +1,330 @@
+          >
+          <b-form-checkbox
+            id="community-only"
+            v-model="commonly"
+            class="d-inline-block ml-2"
+            name="community-only"
+            value="accepted"
+            unchecked-value="not_accepted"
+          >
+          </b-form-checkbox>
+        </div>
+
+        <div class="mt-2">
+          <b-button variant="dark" size="sm" @click="patchMedia"
+            >Update</b-button
+          >
+          <b-button size="sm" variant="dark" class="" @click="toggleModal"
+            >Cancel</b-button
+          >
+        </div>
+      </b-row>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import {
+  getGenericFileType,
+  getYoutubeThumbnail,
+  isValidYoutubeLink,
+  getMediaUrl,
+  getCookie
+} from '@/plugins/utils.js'
+
+const base64Encode = data =>
+  new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(data)
+    reader.onload = () => resolve(reader.result)
+    reader.onerror = error => reject(error)
+  })
+
+export default {
+  components: {},
+  props: {
+    id: {
+      type: Number,
+      default: 0
+    },
+    media: {
+      type: Object,
+      default: () => {
+        return {}
+      }
+    },
+    toggleModal: {
+      type: Function,
+      default: () => {
+        return {}
+      }
+    },
+    showState: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      showModal: false,
+      title: '',
+      description: '',
+      url: null,
+      media_file: null,
+      fileSrc: null,
+      file: null,
+      commonly: 'not_accepted',
+      youtubestate: null
+    }
+  },
+  computed: {
+    hasImgThumbnail() {
+      const type = this.getGenericType
+      return type === 'image' || type === 'youtube'
+    },
+    getGenericType() {
+      return this.getGenericFileType(this.media.file_type)
+    },
+    isPlacename() {
+      return this.$route.query.type || this.$route.query.upload_artwork
+    },
+    isArtwork() {
+      return (
+        this.$route.query.type &&
+        (this.$route.query.type === 'Artist' ||
+          this.$route.query.type === 'Public Art')
+      )
+    },
+    getCookies() {
+      return {
+        headers: {
+          'X-CSRFToken': getCookie('csrftoken')
+        }
+      }
+    }
+  },
+  watch: {
+    file(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        if (newValue) {
+          base64Encode(newValue)
+            .then(value => {
+              this.fileSrc = value
+            })
+            .catch(() => {
+              this.fileSrc = null
+            })
+        } else {
+          this.fileSrc = null
+        }
+      }
+    }
+  },
+  mounted() {
+    const { name, description, community_only, url, media_file } = this.media
+
+    this.title = name
+    this.description = description
+    this.commonly = community_only ? 'accepted' : 'not_accepted'
+    this.showModal = this.showState
+
```

Comment:
Add a comment indicating why these are mutually exclusive?

Reply:
Yeah, that's weird. Can't you just base it on showState instead? Or vice versa.

Reply:
For why `this.showMdal = this.showState` was on the code, it was because vue gave this warning `Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "showState"`